### PR TITLE
Use new Json Pipeline overloads

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Use the new async Json Pipeline overloads introduced in .NET 9.0 https://github.com/dotnet/runtime/pull/101461 to respond better to back pressure on the Json response.

#### What types of changes does your code introduce?

- [x] New feature (a non-breaking change that adds functionality)
- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs [hunk 1]
```diff
@@ -16,16 +16,14 @@ namespace Nethermind.Serialization.Json;
...
-public interface ICountingBufferWriter : IBufferWriter<byte>
+public abstract class CountingWriter : PipeWriter
 {
-    long WrittenCount { get; }
-    ValueTask CompleteAsync(Exception? exception = null);
+    public long WrittenCount { get; protected set; }
 }
 
-public sealed class CountingPipeWriter : ICountingBufferWriter
+public sealed class CountingPipeWriter : CountingWriter
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Serialization.Json/StreamPipeWriter.cs:24]: Why is it better to inherit PipeWriter, when it is clearly a decorator?

File: src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs [hunk 2]
```diff
@@ -143,18 +143,15 @@ private JsonWriterOptions CreateWriterOptions(bool indented)
         public async ValueTask<long> SerializeAsync<T>(Stream stream, T value, bool indented = false, bool leaveOpen = true)
         {
             var writer = GetPipeWriter(stream, leaveOpen);
-            Serialize(writer, value, indented);
+            await JsonSerializer.SerializeAsync(writer, value, indented ? JsonOptionsIndented : _jsonOptions);
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs:146]: Does it make sense to be `ValueTask` method now? Won't that always allocate?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs:146]: `SerializeAsync` returns a `Task`, so it will always allocate?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs:146]: Then this method will allocate an additional Task vs if it synchronously completes it won't?

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs:146]: won't it always allocate as the inner Task is not synchronous?

Comment (benaadams) [reply src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs:146]: The inner task will normally be synchrounous; will only go async when the outbound http buffer is full and it suspends

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Serialization.Json/EthereumJsonSerializer.cs:146]: Why didn't they make it ValueTask then?

**Pull request merged**